gitlabform/processors/project/hooks_processor.py

```diff
@@ -15,52 +15,95 @@ def __init__(self, gitlab: GitLab):
-                    debug("Creating hook '%s'", hook)
-                    created_hook: RESTObject = project.hooks.create(hook_config)
-                    debug("Created hook '%s'", created_hook)
-                elif hook_id and any(diffs):
-                    debug("Changing existing hook '%s'", hook)
-                    changed_hook: Dict[str, Any] = project.hooks.update(
-                        hook_id, hook_config
-                    )
-                    debug("Changed hook to '%s'", changed_hook)
-                elif hook_id and not any(diffs):
-                    debug(f"Hook {hook} remains unchanged")
+                debug(f"Updating hook '{hook}'")
+                updated_hook: Dict[str, Any] = project.hooks.update(
+                    hook_id, hook_config
+                )
+                debug(f"Updated hook: {updated_hook}")
+            else:
+                debug(f"Hook '{hook}' remains unchanged")
 
+        # Process hook config enforcements
         if configuration.get("hooks|enforce"):
-            for gh in hooks_list:
-                if gh.url not in config_hooks:
+            for gh in project_hooks:
+                if gh.url not in hooks_in_config:
                     debug(
-                        "Deleting hook '%s' currently setup in the project but it is not in the configuration and enforce is enabled",
-                        gh.url,
+                        f"Deleting hook '{gh.url}' currently setup in the project but it is not in the configuration and enforce is enabled"
                     )
                     project.hooks.delete(gh.id)
+
+    def is_hook_config_different(
+        self, config_in_gitlab: dict, config_in_gitlabform: dict
+    ):
+        """
+        Compare two dictionary representing a webhook configuration and determine
+        if they are different.
+
+        GitLab's webhook data does not contain "token" as it is considered a secret.
+        If GitLabForm config for a webhook contains a "token", the difference cannot
+        be validated. In this case, the config is assumed to be different. Otherwise,
+        the config data will be compared and determined if they are different.
+
+        Args:
+            config_in_gitlab (dict): hook configuration in gitlab
+            config_in_gitlabform (dict): hook configuration in gitlabform
+
+        Returns:
+            boolean: True if the two configs are different. Otherwise False.
+        """
+        if "token" in config_in_gitlabform:
+            debug(
+                f"The hook '{config_in_gitlabform['url']}' config includes a token. Diff between config vs gitlab cannot be confirmed."
+            )
+            return True
+
+        diffs = (
+            map(
+                lambda k: config_in_gitlabform[k] != config_in_gitlab[k],
+                config_in_gitlabform.keys(),
+            )
+            if config_in_gitlab
+            else iter(())
+        )
+
+        if any(diffs):
```

Comment:
It's not nitpicking. I'm glad you mentioned it. I do tend to forget about python style syntax. Thanks for pointing this out. It's small, simple, and easier to read instead of going through 4 lines.